spec/models/library_entry_spec.rb

```diff
@@ -133,6 +133,52 @@
     end
   end
 
+  describe 'timestamp validation' do
```

Comment:
for all the tests, it might be better to have like a `let!(:library_entry) { create(:library_entry, media: anime) }` and then just do something `LibraryEntry.last` to find it and just continue working with that, instead of creating more and more library entries. I don't think any of your tests conflict with each other that if you change the status of 1 of them, or change the progress you will get different results than expected.

Also just thought i'd explain this to you, 
```
  library_entry = build(:library_entry, media: anime, status: :current)
  library_entry.save!
```
can just be consolidated into `library_entry = create(:library_entry, media: anime, status: :current)` for the first instance (create saves it also)